staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit_test.go

```diff
@@ -936,21 +936,24 @@ func TestAuditJson(t *testing.T) {
 
 type fakeRequestContextMapper struct {
 	user *user.DefaultInfo
+	ctx  request.Context
 }
 
 func (m *fakeRequestContextMapper) Get(req *http.Request) (request.Context, bool) {
-	ctx := request.NewContext()
+	if m.ctx == nil {
+		m.ctx = request.NewContext()
```

Comment:
this makes fakeRequestContextMapper non-thread safe

Reply:
But this is just test, for that particular usecase. I doubt `fakeRequestContextMapper` will be use anywhere else. Same for the next call, as well.